AnnService/inc/Core/SPANN/ExtraFullGraphSearcher.h

```diff
@@ -105,10 +105,14 @@ namespace SPTAG
                 do {
                     auto curIndexFile = f_createAsyncIO();
                     if (curIndexFile == nullptr || !curIndexFile->Initialize(curFile.c_str(), std::ios::binary | std::ios::in, 
+#ifndef _MSC_VER
 #ifdef BATCH_READ
                         p_opt.m_searchInternalResultNum, 2, 2, p_opt.m_iSSDNumberOfThreads
 #else
                         p_opt.m_searchInternalResultNum * p_opt.m_iSSDNumberOfThreads / p_opt.m_ioThreads + 1, 2, 2, p_opt.m_ioThreads
+#endif
+#else
+                        (p_opt.m_searchPostingPageLimit + 1) * PageSize, 2, 2, p_opt.m_ioThreads
```

Comment:
This is a lot cleaner than my change was :)

And good idea to make BATCH_READ explicitly non-MSVC only